enhancements/windows-containers/disconnected-support.md

```diff
@@ -0,0 +1,439 @@
+limited to the Windows Machine Config Operator and its sub-component, Windows Instance Config Daemon (WICD).
+First, WMCO will set the `config_path` variable in the containerd config file.
+Next, the operator will generate registry config files based on cluster resources and publish them to WICD. 
+And lastly, WICD will consume the generate registry config and create them on each Windows instance.
+
+WMCO will use the mirrors data in `ImageDigestMirrorSet` and `ImageTagMirrorSet` resources 
+to create a set of [hosts.toml files](https://github.com/containerd/containerd/blob/main/docs/hosts.md#hoststoml-content-description---detail),
+one for each registry mirror, that will be transferred to the Windows instance. WMCO will handle the file content 
+creation and will publish the required info to WICD. WICD will consume the given info and create/update files on the
+Windows instances.
+
+Since WMCO is an day 2 operator (optional, not part of the OCP payload), it will pick up the mirror configuration during
+runtime regardless of whether they were set during cluster install time or at some point during the cluster's lifetime.
+
+#### Setting `config_path`
+
+WMCO will need to set the `config_path` variable in the containerd config file to tell it where to look for registry
+config settings; it is currently empty in value. We should likely set it to something within our managed directories
+(e.g. "C:\\k") to ensure they're cleaned up during node deconfiguration.
+
+#### File Content
+
+WMCO will generate the required registry config files based on the IDMS/ITMS resources in the cluster.
+The operator will react to updates to the cluster's mirror settings through a new "registry controller" which will
+watch for events on IDMS/ITMS resources.
+WMCO will publish the generated file content into the windows-services ConfigMap.
+Because the windows-services ConfigMap is immutable, publishing changes to it will require a couple steps.
+
+The file content management flow will be as such:
+1. First, we'll introduce a method to update the configmap_controller's `servicesManifest` object which holds the
+  windows-services CM data
+2. then, generate an updated set of hosts.toml files
+3. next, compute a diff between the `Files` in the existing windows-services ConfigMap and the newly generated files
+  - any files in the existing spec that are NOT in the new set will will be added to the serviceManifest's `FilesToDelete`
+  using the method introduced in step 1
+4. update the serviceManifest's `Files` to hold the newly generated set of hosts.toml files
+5. lastly, publish the new windows-services ConfigMap
+  - Delete the cluster's existing windows-services ConfigMap from the registry controller
+  - The configmap_controller will react by re-creating the windows-services ConfigMap (as it does today), with new spec
+
+The windows-services ConfigMap spec will have to be extended. We will add a mandatory `content` field to the `files` key
```

Comment:
Is this done for the CNI config? If not, why is this needed for the containderd / hosts.toml files?